tests/js/client/communication/test-communication.js

```diff
@@ -229,6 +234,19 @@ function CommunicationSuite () {
       // run the suite for 5 minutes
       runTests(tests, 5 * 60);
     },
+
+    // four writers creating lock contention on the same shards / documents
+    testStreamTrxInParallel: function() {
+      let tests = [
+        [ 'stream-trx-1', 'let t = db._createTransaction({ collections: { exclusive: "UnitTestsExc" } }); let tc = t.collection("UnitTestsExc"); let doc = tc.document("entityName"); tc.update(doc._key, {nextOID: doc.nextOID + 1}); t.commit();'],
+        [ 'stream-trx-2', 'let t = db._createTransaction({ collections: { exclusive: "UnitTestsExc" } }); t.query(\'LET d = DOCUMENT("UnitTestsExc/entityName") UPDATE d WITH {nextOID: d.nextOID + 1} IN UnitTestsExc\'); t.commit();'],
+        [ 'stream-trx-3', 'let t = db._createTransaction({ collections: { exclusive: "UnitTestsExc" } }); let tc = t.collection("UnitTestsExc"); for (i = 0; i < 50; i++) {tc.save({nextOID: 0})} t.commit();'],
+        [ 'stream-trx-4', 'let t = db._createTransaction({ collections: { exclusive: "UnitTestsExc" } }); t.query(\'FOR d IN UnitTestsExc UPDATE d WITH {nextOID: d.nextOID + 1} IN UnitTestsExc\'); t.commit();']
+      ];
+
+      // run the suite for 1 minutes
+      runTests(tests, 1 * 60);
```

Comment:
Does this test-suite really catch what you are trying to get here?
This PR limits the max lock time to 30s (or 60s see comments above) this test code tries to perform as many of the above operations as possible, and then tries to shut-down the clients.
In the shutdown process it will wait for at least 30s in worst-case.

So even if the above queries get into a "dead-lock" and are blocking for <maxLocktime> now this test will succeed, and not operation has taken place.

So this test checks if we will eventually get unlocked.
It will not check if we still get into a locked situation.
Can we somehow track the time spend in locks (or spend within the transaction), and assert that it is at least close to 1 * 60 on exclusive case. And above 1 * 60 on write only case? This will still not be perfect as locking here is still a game of chance, but it will eventually hit.

Reply:
my assumption was that if an arangosh instance gets into a dead-lock then it will not terminate by itself within the 30s time-frame allowed by the `runTests(...)` function. 
This should then manifest as an error of this testsuite